packages/react-router-dom/index.tsx

```diff
@@ -351,13 +351,9 @@ if (__DEV__) {
 export { ViewTransitionContext as UNSAFE_ViewTransitionContext };
 
 // TODO: (v7) Change the useFetcher data from `any` to `unknown`
-type FetchersContextObject = {
-  fetcherData: Map<string, any>;
-  register: (key: string) => void;
-  unregister: (key: string) => void;
-};
+type FetchersContextObject = Map<string, any>;
 
-const FetchersContext = React.createContext<FetchersContextObject | null>(null);
+const FetchersContext = React.createContext<FetchersContextObject>(new Map());
```

Comment:
The ref counting now lives in the router so this fetchers context just holds the data